### PR TITLE
Adding team view

### LINKED ISSUE #94: Add team view
The team view functionality should be added, along with the necessary HTML files, to properly present team divisions

### PR DESCRIPTION
**Pull request by rotem2022**
### What this PR does?

In this PR I added generate team button
added files:

1. event/migrations/0007_static_data.py :

- Create users and events to facilitate testing the "Generate Team" button.
- This migration ensures that the necessary data is available for testing.

edit files:

1. event/templates/event/event_info.html : 

- Added a button to the file that allows generating teams for the event.
- Note that the generation can only occur once, as agreed upon.
- After clicking the button, a table displaying the generated teams will be shown.

2. event/urls.py :
- Include a relevant path for the newly added functionality.

3. event/views.py
- Make necessary changes to handle the team generation button functionality.
4. teams/models.py
- Implement any modifications required for the team generation feature.
5. teams/tests.py:
- Updated the test to ensure the correctness of the team generation functionality.

### Related Issue 

Resolves #94 

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

teams3
teams button

### CODE REVIEW

**Review by Sendy459 — APPROVED** [review]

**Review by Sendy459 — DISMISSED** [review]

**Review by Ofekmn — APPROVED** [review]

**Review by Sendy459 — DISMISSED** [review]

**Review by Sendy459 — DISMISSED** [review]

**Review by Ofekmn — DISMISSED** [review]

**Review by rimonsh — DISMISSED** [review]

File: event/templates/event/event_info.html [hunk 1]
```diff
@@ -12,6 +12,63 @@ <h2>{{event.name}} Event</h2>
         <li>Participants : {{event.participants_num}}/{{event.max_participants}}</li>
     </ul>
 </div>
+    {% if team1|length == 0%}
+        <form id="event-form" method="post">
+
+        {% csrf_token %} {{ form.as_p }}
+    <button type="submit">Generate Teams</button>
```
Comment (ygalblum) [inline event/templates/event/event_info.html:19]: I'm not an HTML guy, but, is the indentation supposed to look like that? For example, `form` starts in 8 spaces but ends in 0

File: event/views.py [hunk 2]
```diff
@@ -42,7 +43,26 @@ def create_event(request, user_id):
...
+    team2 = []
+    users_events = UserEvent.objects.filter(eventID=event_id)
+    if len(users_events) > 0:
+        if users_events[0].teamID:
+            teams = list(set([user_event.teamID
+                              for user_event in users_events]))
+            team1 = [user.userID.user.username
+                     for user in UserEvent.objects.filter(teamID=teams[0].id)]
+            team2 = [user.userID.user.username
+                     for user in UserEvent.objects.filter(teamID=teams[1].id)]
```
Comment (ygalblum) [inline event/views.py:59]: Combining the two conditions together allows you to go back one indentation
```suggestion
    if len(users_events) > 0 and users_events[0].teamID:
        teams = list(set([user_event.teamID for user_event in users_events]))
        team1 = [user.userID.user.username for user in UserEvent.objects.filter(teamID=teams[0].id)]
        team2 = [user.userID.user.username for user in UserEvent.objects.filter(teamID=teams[1].id)]
```

File: event/views.py [hunk 3]
```diff
@@ -42,7 +43,26 @@ def create_event(request, user_id):
...
-    return render(request, 'event/event_info.html', {'event': event})
+    if request.method == "POST":
+        Teams.generate_teams(event_id)
+
+    team1 = []
+    team2 = []
+    users_events = UserEvent.objects.filter(eventID=event_id)
+    if len(users_events) > 0:
+        if users_events[0].teamID:
+            teams = list(set([user_event.teamID
```
Comment (ygalblum) [inline event/views.py:54]: You check `users_events[0].teamID`, and then run on the entire list. Is it guarantied that the rest of the items have it?

File: event/views.py [hunk 4]
```diff
@@ -42,7 +43,26 @@ def create_event(request, user_id):
...
+                              for user_event in users_events]))
+            team1 = [user.userID.user.username
```
Comment (ygalblum) [inline event/views.py:56]: How are you sure that `len(teams) == 2`?

File: teams/models.py [hunk 5]
```diff
@@ -17,11 +17,18 @@ def generate_teams(event_id):
         from event.models import UserEvent
         users_events = UserEvent.objects.filter(eventID=event_id)
         team_size = len(users_events) // 2
-        team1 = Teams(name="Team1")
+        team1 = Teams(name=f"{event_id}Team1")
```
Comment (ygalblum) [inline teams/models.py:20]: Consider a seperator between the event_id and Team. 
```suggestion
        team1 = Teams(name=f"{event_id}-Team1")
```

**Review by ygalblum — CHANGES_REQUESTED** [review]

File: event/templates/event/event_info.html (hunk 1 above)
Comment (rotem2022) [reply event/templates/event/event_info.html:19]: I'm not sure I understand your meaning, can you please explain?

File: event/views.py (hunk 2 above)
Comment (rotem2022) [reply event/views.py:59]: This implementation is structured in this particular way due to the scenario where the length of the `user_event` is 0. In such cases, attempting to access `user_event[0].teamID` would result in an error.

File: event/views.py (hunk 3 above)
Comment (rotem2022) [reply event/views.py:54]: Yes it is guarantied. If a user belonged to a team, it means that the teams are generated.
Therefore, it is guarantied there are 2 teams to the event.

File: event/views.py (hunk 4 above)
Comment (rotem2022) [reply event/views.py:56]: same as the above

**Review by Sendy459 — DISMISSED** [review]

File: event/templates/event/event_info.html (hunk 1 above)
Comment (ygalblum) [reply event/templates/event/event_info.html:19]: Indentation help read the document. So, I would expect that opening and closing a tag will happen in the same indentation. For example:
This is good:
```
<form>
</form>
```
This is not so much:
```
    <form>
        </form>
```

I think the issue actually starts a line before (the location of the if statement). So, it should be:
```
{% if team1|length == 0 %}
<form id="event-form" method="post">

    {% csrf_token %} {{ form.as_p }}
    <button type="submit">Generate Teams</button>

</form>
{% else %}
...
```

File: event/views.py (hunk 3 above)
Comment (ygalblum) [reply event/views.py:54]: OK

File: event/views.py (hunk 4 above)
Comment (ygalblum) [reply event/views.py:56]: OK

File: event/templates/event/event_info.html (hunk 1 above)
Comment (rotem2022) [reply event/templates/event/event_info.html:19]: Ok, thank you!

**Review by SaarOrus — APPROVED** [review]

**Review by ygalblum — APPROVED** [review]

**Pull request merged**